ArduCopter/GCS_Mavlink.cpp

```diff
@@ -1025,6 +1020,34 @@ MAV_RESULT GCS_MAVLINK_Copter::handle_command_pause_continue(const mavlink_comma
     return MAV_RESULT_DENIED;
 }
 
+MAV_RESULT GCS_MAVLINK_Copter::handle_command_nav_takeoff(const mavlink_command_int_t &packet)
+{
+    // takeoff altitude in centimeters above home
+    int32_t takeoff_alt;
+
+    // create takeoff location from command int
+    Location takeoff_loc;
+    if (!location_from_command_t(packet, takeoff_loc)) {
+        return MAV_RESULT_FAILED;
+    }
+
+    // sanitize the location for terrain frame
+    takeoff_loc.lat = copter.current_loc.lat;
+    takeoff_loc.lng = copter.current_loc.lng;
+
+    // get takeoff altitude above home
+    if(!(takeoff_loc.get_alt_cm(Location::AltFrame::ABOVE_HOME, takeoff_alt))){
+        return MAV_RESULT_FAILED;
+    }
+
+    // try to takeoff
+    if (!copter.flightmode->do_user_takeoff(takeoff_alt, is_zero(packet.param3))) {
```

Comment:
Let's remove the param3 usage completely
```suggestion
    if (!copter.flightmode->do_user_takeoff(takeoff_alt)) {
```

Reply:
@hendjoshsr71 from everywhere? Because there is no implementation like `do_user_takeoff(takeoff_alt)`...